File: tests/fixtures/code_meta/good.ts
// @chunk
// id: code:demo.fuse
// type: function
// implements: page:hybrid-rag
// tags: [rrf]
// summary: Fuse two rankings
// @end
export function fuse(a: string[], b: string[]): string[] {
  return [...a, ...b]
}

/* @chunk
 * id: code:demo.note
 * type: const
 * summary: Tiny helper
 * @end
 */
export const note = "ok"

// @chunk id=code:demo.tiny type=function
export function tiny(): number {
  return 2
}

File: tests/fixtures/code_meta/good.py
# @chunk
# id: code:demo.hybrid_search
# type: function
# implements: page:hybrid-rag
# citations: [hm-rag-2025]
# tags: [retrieve, rrf]
# summary: RRF fuse of BM25 and dense
# @end
def hybrid_search(query: str) -> list[str]:
    return [query]


# @chunk id=code:demo.tokenize type=function tags=[tokenize]
def tokenize(text: str) -> list[str]:
    return text.split()


def unmarked_helper() -> int:
    return 1
